redis/client.py

```diff
@@ -3606,21 +3626,21 @@ def execute_command(self, *args):
         kwargs = {'check_health': not self.subscribed}
         self._execute(connection, connection.send_command, *args, **kwargs)
 
-    def _execute(self, connection, command, *args, **kwargs):
-        try:
-            return command(*args, **kwargs)
-        except (ConnectionError, TimeoutError) as e:
-            connection.disconnect()
-            if not (connection.retry_on_timeout and
-                    isinstance(e, TimeoutError)):
-                raise
-            # Connect manually here. If the Redis server is down, this will
-            # fail and raise a ConnectionError as desired.
-            connection.connect()
-            # the ``on_connect`` callback should haven been called by the
-            # connection to resubscribe us to any channels and patterns we were
-            # previously listening to
-            return command(*args, **kwargs)
+    def _disconnect_raise_connect(self, conn, error):
+        conn.disconnect()
+        if not (conn.retry_on_timeout and isinstance(error, TimeoutError)):
+            raise error
+        conn.connect()
+
+    def _execute(self, conn, command, *args, **kwargs):
+        # Connect manually upon disconnection. If the Redis server is down,
```

Comment:
This looks like it should be a docstring rather than a line comment.

Reply:
@abrookins thanks, resolved. Please note I simply moved the comment from the previous location in the code when I moved the code to this new function.